### PR TITLE
Add tox and GitHub Actions

### PR DESCRIPTION
**Pull request by eumiro**
Add tox and GitHub Actions (along with Python 3.8 and 3.9). Since this repository still does not use GHA in its main branch, the badge won't be shown immediately, but the CI results can be seen in my branch.

### CODE REVIEW

**Comment by eumiro** [comment]
Oh yes, it looks like it has worked: https://github.com/eumiro/destream/runs/1674539754

File: .github/workflows/ci.yml
Comment (cecton) [inline .github/workflows/ci.yml:25]: Missing EOF newline (sorry for that)

File: .gitignore
Comment (cecton) [inline .gitignore:2]: Revert this. I'm not sure why it's there but I suspect there is a good reason. I know why you removed it: you wanted to add the .github but you can do that with `git add -f` or you can add an exception `!/.github` in the .gitignore instead

File: setup.py
Comment (cecton) [inline setup.py:16]: Awesome 👍

File: tox.ini
Comment (cecton) [inline tox.ini:16]: Missing EOF newline (sorry again 😁 )

**Review by cecton — CHANGES_REQUESTED** [review]
Excellent work!! Thanks a lot

**Comment by cecton** [comment]
Do you want to join the owners/maintainers? It would help reduce the bus factor (I'm here temporarily)

File: .github/workflows/ci.yml
Comment (eumiro) [reply .github/workflows/ci.yml]: Thanks. That's a good reason to add `pre-commit`.

File: .gitignore
Comment (eumiro) [reply .gitignore:2]: That may be because of some `.*` dirs/files added by IDEs. Okay, I have reverted it and added the `!/.github` line.

File: tox.ini
Comment (eumiro) [reply tox.ini]: Fixed, thanks.

**Comment by eumiro** [comment]
> Do you want to join the owners/maintainers? It would help reduce the bus factor (I'm here temporarily)

I came because EuroPython retweeted your request and saw there's work to be done. You can add me to the list.

File: .github/workflows/ci.yml
Comment (cecton) [reply .github/workflows/ci.yml]: Yeah haha 😂  you referencing to that conversation on twitter?

File: .gitignore
Comment (cecton) [reply .gitignore:2]: I'm not sure. I don't use an IDE I use ViM. But it's not impossible that I did that for my colleagues back in the days. If you become maintainer feel free to remove it but remember there might be a reason it was there

**Comment by cecton** [comment]
@eumiro it's weird that we don't see the checks running on this PR. We should see them. I think you might have something messed up with the triggers.

I usually use this:

```yaml
on:
  pull_request:
    branches: [main]
  push:
    branches: [main]
  schedule:
    - cron: 0 0 1 * *
```

So it builds:
- when there is a pull request against the branch main
- when there is a push on the branch main
- once every month (to make sure it is stable)

**Comment by eumiro** [comment]
I'll change it to what you proposed (actually wanted to add that cron entry to my projects as well), but this won't help anyway, because I believe that if `.github/workflows` is not in the upstream repository, it won't be shown. And this is the very first PR introducing the CI.

And my current proposal is just more general, because it triggers at any push/PR and not only against the `main` branch.

**Comment by eumiro** [comment]
Now I'll need a review approval and we can go on.

**Review by cecton — APPROVED** [review]
OK let's see if it works \o/

**Pull request merged**
